### PR TITLE
py-vermin: add latest version 1.5.2

### PR DESCRIPTION
**Pull request by netromdk**
Released version [1.5.2](https://github.com/netromdk/vermin/releases/tag/v1.5.2) with various fixes and new rules.

@adamjstewart @alalazo

### CODE REVIEW

**Review by adamjstewart — COMMENTED** [review]
Is setuptools actually used at run-time? I don't see it used anywhere. Can also remove the python 2.7+ constraint as Spack only supports installing 3.7+ anyway.

**Comment by netromdk** [comment]
No setuptools are not used at runtime and py27 can be removed, too. I'll do that.

File: var/spack/repos/builtin/packages/py-vermin/package.py [hunk 1]
```diff
@@ -37,9 +38,6 @@ class PyVermin(PythonPackage):
     version("0.10.4", sha256="bd765b84679fb3756b26f462d2aab4af3183fb65862520afc1517f6b39dea8bf")
     version("0.10.0", sha256="3458a4d084bba5c95fd7208888aaf0e324a07ee092786ee4e5529f539ab4951f")
 
-    depends_on("python@2.7:", type=("build", "run"))
-    depends_on("py-setuptools", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-vermin/package.py:41]: It's still needed to build the package though:
```python
depends_on("py-setuptools", type="build")
```
Were there any older versions that required it at run-time?

Comment (netromdk) [reply var/spack/repos/builtin/packages/py-vermin/package.py:41]: Oh, yeah, my bad. No, I think I incorrectly copied it from another recipe initially.

**Comment by netromdk** [comment]
Should I actually include something like the following?
Or did you mean that 3.7+ is always available via Spack?
```py
depends_on("python@3:", type=("run"))
```
Or some syntax for 3.x, I mean.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**